### DISCUSSION TITLE
Right click, lock results tab.

### DISCUSSION THREAD

**Discussion by gtdegamo** [dbeaver/dbeaver#14451] [Q&A] [answered]
**Is your feature request related to a problem? Please describe.**

More often than not, I overwrite the results of a results tab b/c I'm too focused on the current query I am trying to make.

It's usually not so bad b/c the query which made the lost result set is in the same console window.  So I can just rerun the query.  But, sometimes I don't have it, and I lose that work.  I don't often save these queries, because they are temporary, but they took me some time to iron out.  Or many times, the query that I have saved takes several minutes to reproduce.

**Describe the solution you'd like**

After a result tab appears with the results of a query.  We should be able to right click the result tab and just mark the tab as keep/locked for session.  You should be able to toggle it off in the same manner.

Then, in the sql console, if you press 'play' and that will  overwrite the current locked result set, dbeaver will warn you:

'Cannot overwrite locked result set.  Either press play in another tab or unlock the result set.'

Now have an option to 'close all unlocked result tabs'.  Or maybe modify the behavior of 'close all result tabs' so that it doesn't kill the locked results.  Finally, if you implement the 'close result tabs to the right/left', the close operation should skip the locked result tabs.

**Describe alternatives you've considered**

No alternative but to recreate/rerun the query to reproduce the lost results.

**Additional context**

None.  Keep up the good work.

**Comment by p91paul** [comment]
there should already be this feature. I can't remember if it's called lock result tab or pin result tab, but it works like you described. Running a new query will automatically open a new tab.
Also, I can add you can use the query manager (view -> windows) to see all the query you've run in the session if you need recovering one.

**Comment by ShadelessFox** [comment, marked as answer]
Hello @gtdegamo,

You can "lock" tab by using <kbd>Pin tab</kbd> action accessible via context menu.

Please let me know if this is what you want.
